src/Symfony/Component/Form/Extension/Core/Type/ImmutableChoiceType.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Base class for child of {@link ChoiceType}.
+ *
+ * Every class extending it is responsible for loading
+ * the choices and eventually cache them.
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+abstract class ImmutableChoiceType extends AbstractType
```

Comment:
I think the benefits of adding this abstract class don't outweigh the disadvantages of a more complex class inheritance tree. Usually composition in types should be preferred over inheritance, since inheritance is much less flexible.

For example, if you add a protected method and later find out you need another argument in that method, your limited by our BC policy. Creating a new object instance within a method body is not restricted by that policy and much more flexible.